HW_TypeScript_Primitive_literal_types_and_object_typing.ts

```diff
@@ -0,0 +1,194 @@
+class School {
+  _areas: Area[] = [];
+  _lecturers: {
+    name: string;
+    surname: string;
+    position: string;
+    company: string;
+    experience: number;
+    courses: string[];
+    contacts: string[];
+    fullName: string;
+  }[] = [];
+
+  addArea(area: Area): void {
+    this._areas.push(area);
+  }
+
+  removeArea(areaName: string): void {
+    this._areas = this._areas.filter(area => area.name !== areaName);
+  }
+
+  addLecturer({ name, surname, position, company, experience, courses, contacts }: {
+    name: string,
+    surname: string,
+    position: string,
+    company: string,
+    experience: number,
+    courses: string[],
+    contacts: string[]
+  }): void {
+    const lecturer = {
+      name,
+      surname,
+      position,
+      company,
+      experience,
+      courses,
+      contacts,
+      fullName: `${surname} ${name}`
+    };
+    this._lecturers.push(lecturer);
+  }
+
+  get areas() {
+    return this._areas;
+  }
+
+  get lecturers() {
+    return this._lecturers;
+  }
+}
+
+class Area {
+  _levels: Level[] = [];
+  _name: string;
+
+  constructor(name: string) {
+    this._name = name;
+  }
+
+  get name(): string {
+    return this._name;
+  }
+}
+
+class Level {
+
+  _groups: Group[] = [];
+  _name: string;
+  _description: string;
+
+  constructor(name: string, description: string) {
+    this._name = name;
+    this._description = description;
+  }
+
+  addGroup(group: Group) {
```

Comment:
Please, do not forget to add explicitly type annotation for the return function type.